### ISSUE TITLE
[Question] How to tell purgecss to NOT scan a subfolder within the specified folder?

### ISSUE THREAD

**Issue by raniesantos** [FullHuman/purgecss#158]
In a project structure where I have this:

- source/
  - foo/
  - bar/
    - dontscan.ext
  - index.html

I specified this in the config `folders: ['source']`.

How do I configure the plugin to NOT scan for words in the `bar` directory?

**Comment by ThisNameWasTaken** [comment]
The [options for purgecss](https://www.purgecss.com/configuration#options) look like this:
```js
{
  content: Array<string | RawContent>,
  css: Array<string | RawContent>,
  extractors?: Array<ExtractorsObj>,
  whitelist?: Array<string>,
  whitelistPatterns?: Array<RegExp>,
  whitelistPatternsChildren?: Array<RegExp>,
  keyframes?: boolean,
  fontFace?: boolean,
  rejected?: boolean
}
```
As you can see there is no option called `folders`.

If you want to scan all the content inside a given folder you need to pass a [glob](https://www.npmjs.com/package/glob#glob-primer) to the `content` option.

Here is an example:
```js
content: [
    // keep only those that you need
    '**/*.html' // will look inside any folder for an .html file
    '!(bar)/**/*.html' // will look inside any folder that is not 'bar'
    '!({bar,foo})/**/*.html' // will look inside any folder that is nor 'bar' nor 'foo'
    '!(bar-*)/**/*.html' // will look inside any folder that does not start with 'bar-'
  ],
  css: [`**/*.css`],
```

**Comment by bdkjones** [comment]
@ffloriel Has this behavior changed? Using PurgeCSS 4.0.0, I'm trying to skip the `node_modules` folder using:

```
content : [
    '!(node_modules)/**/*.html
]
```

The project structure:
```
source
    index.html
    css
        main.css
    node_modules
    ...
```

This fails. PurgeCSS removes ALL rules from the stylesheet (as if it's not scanning `index.html`).

If I use:
```
content : [
    '**/*.html
]
```
Purging works correctly.

Additionally, it seems that the "negation" operator cannot appear in the middle of a path. For example, this fails:

`some/folder/!(node_modules)/**/*.html`

**Comment by bdkjones** [comment]
After some digging, it looks like the glob dependency you use HAS changed this behavior:


![Screen Shot 2021-02-28 at 02 56 17](https://user-images.githubusercontent.com/926566/109416009-a6ff9d80-7970-11eb-8aed-c91d5d51194e.png)

Since it's no longer possible to negate parts of the globs, I'll open a new issue to request support for the globbing `ignore` parameter.

**Comment by bdkjones** [comment]
For anyone who stumbles on this from Google (as I did), the above advice from @ThisNameWasTaken is no longer valid. Changes to the dependencies that PurgeCSS uses have removed support for the `!` operator in globs. 

There is not (as of March 2021) currently a way to tell PurgeCSS to skip certain folders. I have requested that PurgeCSS return support for this in this issue: https://github.com/FullHuman/purgecss/issues/620
